js/web/lib/wasm/jsep/webgpu/gpu-data-manager.ts

```diff
@@ -45,6 +45,11 @@ export interface GpuDataManager {
    * actually released.
    */
   refreshPendingBuffers(): void;
+
+  /**
+   * destroy all gpu buffers. Call this when the session.release is called.
+   */
+  dispose(): void;
```

Comment:
Currently, for buffer reusing, all storage buffers are not really destroyed. So I want to provide an extra dispose interface to really destroy all allocated gpu resources, which can be triggered when session.release is called in future.

Currently, session is not exposed in backend_webgpu.ts. I am wondering whether we can expose  the session object. All sessions share the webgpu backend. 
one webgpu backend contains multiple sessions. One session contains one gpuDataManager to manage this session's data. When this session is released, all underlying gpu resources will be destroyed. In the session, the buffer can be reused.